### PR TITLE
8261496: Shenandoah: reconsider pacing updates memory ordering

### PR DESCRIPTION
**Pull request by shipilev**
This backport does not apply cleanly, because the `Atomic` parameter orders had changed since [JDK-8234563](https://bugs.openjdk.java.net/browse/JDK-8234563). Passes `hotspot_gc_shenandoah`.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8261496](https://bugs.openjdk.java.net/browse/JDK-8261496): Shenandoah: reconsider pacing updates memory ordering

### Reviewers
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - Committer)
 * [Christoph Langer](https://openjdk.java.net/census#clanger) (@RealCLanger - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk11u-dev pull/66/head:pull/66` \
`$ git checkout pull/66`

Update a local copy of the PR: \
`$ git checkout pull/66` \
`$ git pull https://git.openjdk.java.net/jdk11u-dev pull/66/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 66`

View PR using the GUI difftool: \
`$ git pr show -t 66`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk11u-dev/pull/66.diff">https://git.openjdk.java.net/jdk11u-dev/pull/66.diff</a>

</details>

### CODE REVIEW

**Review by rkennke — APPROVED** [review]

**Comment by shipilev** [comment]
Got the push approval.

/integrate

**Review by RealCLanger — APPROVED** [review]
Approving when @rkennke approves 😄 I think we have to vote him up as official jdk-updates-reviewer.

**Pull request closed without merging**
